src/Microsoft.Health.Dicom.SqlServer/Exceptions/InvalidSchemaVersionException.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Microsoft.Health.Dicom.SqlServer.Exceptions;
 
-internal class InvalidSchemaVersionException : DicomServerException
+internal class InvalidSchemaVersionException : DataStoreNotReadyException
```

Comment:
There are really two cases this happens:
(1) we hecked up and there isn't a data store that supports this version of the schema OR
(2) the DB hasn't been initialized

Should we change the schema version cache to throw `InvalidSchemaVersionException` (or perhaps directly throw a `DataStoreNotReadyException` in case (2) and a regular ol' `KeyNotFoundException` for case (1)?